Updated itree help text to include warning about large collections (main)

korydraughn opened the pull request:
Rewrote help text to match look of `itouch` as well.

trel commented:
Seems good - consider reporting what a memory exhaustion would look like to the user here - so they could identify if they hit it?

Not sure if that's the best idea...

alanking reviewed (commented):
Seems good.

Question: Are we planning to lean on `boost::program_options` (or some equivalent) to generate help text in the future for us as was being done here, or are we going to format the text ourselves? Just trying to understand the school of thought there for the future.

korydraughn commented:
Let's format the text ourselves. We lose control by relying on Boost. Also, if we ever decide to switch to a different library for handling program options, then we know the help text will be unaffected.

korydraughn commented:
> Seems good - consider reporting what a memory exhaustion would look like to the user here - so they could identify if they hit it?
> 
> Not sure if that's the best idea...

Did you have something in mind? I'm not sure what that would look like.

korydraughn commented on src/itree.cpp:
```diff
@@ -340,7 +338,42 @@ auto correct_path(const po::variables_map& pm, rodsEnv& env) -> fs::path {
+                  Filter data objects by a regular expression.
+  -x, --regex-extended-syntax
+                  Enable extended syntax for regular expressions.
+  -s, --size      Display the size of each data object.
```
Done.

korydraughn commented:
I've adjusted the help text again to not use "memory exhaustion".

I've also removed the USAGE print out logic when an exception occurs.

Any thoughts/concerns before I squash and pound?

trel commented on src/itree.cpp:
```diff
@@ -340,7 +337,41 @@ auto correct_path(const po::variables_map& pm, rodsEnv& env) -> fs::path {
+  -C, --color     Print in color (requires ANSI terminal).
+  -f, --fullpath  Print the full path of each item.
+  -F, --classify  Display a forward slash (/) at the end of collections.
+  -i, --ignore-regex=PATTERN
```
-h help?

korydraughn replied:
Done.

alanking reviewed (commented):
Seems good to me. Please squash and leave off #.

korydraughn commented:
Squashed without pound.

korydraughn commented:
Adding pound.
